addons/website_event_track/static/src/interactions/event_track_reminder.js

```diff
@@ -1,87 +1,74 @@
-import { debounce } from "@web/core/utils/timing";
-import publicWidget from "@web/legacy/js/public/public_widget";
 import { _t } from "@web/core/l10n/translation";
 import { rpc } from "@web/core/network/rpc";
 import { Component } from "@odoo/owl";
+import { Interaction } from "@web/public/interaction";
+import { registry } from "@web/core/registry";
 
-publicWidget.registry.websiteEventTrackReminder = publicWidget.Widget.extend({
-    selector: '.o_wetrack_js_reminder',
-    events: {
-        'click': '_onReminderToggleClick',
-    },
+export class WebsiteEventTrackReminder extends Interaction {
+    static selector = ".o_wetrack_js_reminder";
 
-    /**
-     * @override
-     */
-    init: function () {
-        this._super.apply(this, arguments);
-        this._onReminderToggleClick = debounce(this._onReminderToggleClick, 500, true);
-        this.notification = this.bindService("notification");
-    },
+    dynamicContent = {
+        _root: { "t-on-click.prevent.stop.withTarget": this.debounced(this.onReminderToggleClick, 500, true) },
+        "i" : { 
+            "t-att-class": () => {
+                if (this.reminderOn === undefined) return {};
+                return { "fa-bell": this.reminderOn, "fa-bell-o": !this.reminderOn };
+            },
+            "t-att-title": (el) => {
+                if (this.reminderOn === undefined) return el.dataset.reminderOn ? _t("Favorite On") : _t("Set Favorite");
+                return this.reminderOn ? _t("Favorite On") : _t("Set Favorite");
+            },
+        },
+    };
 
-    //--------------------------------------------------------------------------
-    // Handlers
-    //-------------------------------------------------------------------------
-
-    /**
-     * @private
-     * @param {Event} ev
-     */
-    _onReminderToggleClick: function (ev) {
-        ev.stopPropagation();
-        ev.preventDefault();
-        var self = this;
-        var $trackLink = $(ev.currentTarget).find('i');
+    async onReminderToggleClick(ev, currentTargetEl) {
+        const trackLinkEl = currentTargetEl.querySelector("i");
 
         if (this.reminderOn === undefined) {
-            this.reminderOn = $trackLink.data('reminderOn');
+            this.reminderOn = trackLinkEl.dataset.reminderOn;
```

Comment:
Can't this be done in `setup()` instead ? Removing the need for all the conditions about it being missing ?